client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -387,19 +388,18 @@ export class DomainDetailsForm extends PureComponent {
 	}
 
 	renderCountryDependentAddressFields( needsOnlyGoogleAppsDetails ) {
+		const { isStateRequiredInAddress } = this.props;
 		return (
 			<div className="checkout__domain-details-country-dependent-address-fields">
-                { ! needsOnlyGoogleAppsDetails && this.renderAddressFields() }
-                { ! needsOnlyGoogleAppsDetails && this.renderCityField() }
-                { ! needsOnlyGoogleAppsDetails && this.renderStateField() }
-                { this.renderPostalCodeField() }
+				{ ! needsOnlyGoogleAppsDetails && this.renderAddressFields() }
+				{ ! needsOnlyGoogleAppsDetails && this.renderCityField() }
+				{ isStateRequiredInAddress && ! needsOnlyGoogleAppsDetails && this.renderStateField() }
+				{ this.renderPostalCodeField() }
```

Comment:
I am not fully aware of what `needsOnlyGoogleAppsDetails` is but to me it sounds like it's not worth trying to squeeze in the logic into our postal address forms, so what about a third component `<GoogleAppsDetailsForm>`?

Reply:
I like the approach. When it yields `true`, `needsOnlyGoogleAppsDetails` results in a very minimal form anyway. If we're creating new components such as `<{Region}AddressForm />`, it might not be a long stretch to create another for this purpose 👍